isisd/isis_sr.c

```diff
@@ -1980,20 +1980,62 @@ static void show_prefix_sids(struct vty *vty, struct isis_area *area, int level)
  * Declaration of new show commands.
  */
 DEFUN(show_sr_prefix_sids, show_sr_prefix_sids_cmd,
-      "show isis segment-routing prefix-sids",
-      SHOW_STR PROTO_HELP
+      "show isis [vrf <NAME|all>] segment-routing prefix-sids",
+      SHOW_STR PROTO_HELP VRF_CMD_HELP_STR
+      "All VRFs\n"
       "Segment-Routing\n"
       "Segment-Routing Prefix-SIDs\n")
 {
-	struct listnode *node;
+	struct listnode *node, *inode, *nnode;
 	struct isis_area *area;
-
-	for (ALL_LIST_ELEMENTS_RO(isis->area_list, node, area)) {
-		vty_out(vty, "Area %s:\n",
-			area->area_tag ? area->area_tag : "null");
-
-		for (int level = ISIS_LEVEL1; level <= ISIS_LEVELS; level++)
-			show_prefix_sids(vty, area, level);
+	struct isis *isis = NULL;
+	char *vrf_name = NULL;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+
+	ISIS_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
```

Comment:
Let's find a way to not cut-n-paste code for the cse of the single vrf -vs- the default vrf

Reply:
Sure, will find a way and do it for the case of a single vrf -vs- the default vrf.